impl/src/main/java/org/jboss/weld/bootstrap/events/ContainerLifecycleEvents.java

```diff
@@ -241,6 +252,9 @@ public void fireProcessBean(BeanManagerImpl beanManager, Bean<?> bean) {
             } else if (bean instanceof ProducerMethod<?, ?>) {
                 ProcessProducerMethodImpl.fire(beanManager, (ProducerMethod<?, ?>) bean);
             } else {
+                if (isProcessSynthethicBeanObserved()) {
```

Comment:
In fact, it might make sense to create a simple wrapper for `Bean<?>` and `Extension`, i.e. `BeanContext` like we do for `SlimAnnotatedType` (see `SlimAnnotatedTypeContext`).